File: cpp/source/problems/compatibility.cpp
#include "compatibility.hpp"

Compatibility::Compatibility(size_t numObjects, size_t numLabels) :
    numObjects(numObjects),
    numLabels(numLabels),
    save(true),
    verbose(2)
{
}

Compatibility2Pairs::Compatibility2Pairs(size_t numObjects, size_t numLabels) :
	Compatibility(numObjects, numLabels),
    compatibility(numObjects, numLabels, numObjects, numLabels)
{
	compatibility.setZero(); // TODO: MGLOVER - OPTIMIZATION - You may want to leave everything default-initialized.
}

Compatibility3Pairs::Compatibility3Pairs(size_t numObjects, size_t numLabels) :
	Compatibility(numObjects, numLabels),
    compatibility(numObjects, numLabels, numObjects, numLabels, numObjects, numLabels)
{
	compatibility.setZero(); // TODO: MGLOVER - OPTIMIZATION - You may want to leave everything default-initialized.
}

File: python/verify-chessboard-corners/compatibility.py
import numpy as np

class Compatibility:
    def __init__(self, numObjects, numLabels):
        self.numObjects = numObjects
        self.numLabels = numLabels
        self.save = True
        self.verbose = 2

class Compatibility2Pairs(Compatibility):
    def __init__(self, numObjects, numLabels):
        self.__super__(numObjects, numLabels)
        self.compatibility = np.zeros(shape = (numObjects,numLabels,numObjects,numLabels), dtype = np.float64)

class Compatibility3Pairs(Compatibility):
    def __init__(self, numObjects, numLabels):
        super(Compatibility3Pairs,self).__init__(numObjects, numLabels)
        self.compatibility = np.zeros(shape = (numObjects,numLabels,numObjects,numLabels,numObjects,numLabels), dtype = np.float64)
